### PR TITLE
Fix some of validation errors

### PR DESCRIPTION
**Pull request by mehdisadeghi**
Shape id generation was incorrect and we had duplicated trip_ids. This PR fixes those problems. I also fixed some time-travel errors by ignoring stops that stretch into the next day. (I wonder how real GTFS feeds designate such trips that start in one day and finish in another day?).

There are still more errors due to missing data which I left there. There are also many warnings that we can fix, some of them are interesting. Routes with only one stop or very close stops that are probably pointing to the same stop.

### CODE REVIEW

**Comment by mehdisadeghi** [comment]
@carocad this is the new validation results. Have a look! #26 

[validation-results.zip](https://github.com/hiposfer/osmtogtfs/files/1557357/validation-results.zip)

**Comment by carocad** [comment]
> (I wonder how real GTFS feeds designate such trips that start in one day and finish in another day?).

AFAIK that is a valid GTFS feed. I dont remember where I read it but I think that if a trip spawn two days then the stop times are suppose to go over 24:00, for example: 25:13 means the day after at 01:13. I have to search for the reference though

> Routes with only one stop or very close stops that are probably pointing to the same stop.

I think we should eliminate the routes with only one stop. Those will create lots of trouble in kamal. 

As for the stops that are very close, I suppose that the problem are the stops that are in different directions. Those are represented differently in OSM thus GTFS "sees" them as noise. I am not sure if we should eliminate those. I think that we can safely leave those there for the time being.

I also saw a lot of unused stops. I think we can leave those there for the time being. Since they dont have any route, they will not be used for routing so we can deal with that problem later.

As a side note, could you make the PR base be the other branches, it gets really difficult to identify what changed since the other changes are also displayed

**Comment by mehdisadeghi** [comment]
> if a trip spawn two days then the stop times are suppose to go over 24:00

Sounds good. I'll give it a try this way.

> could you make the PR base be the other branches

Good to know that a PR base can be another branch! I changed it.

File: tests/out/shapes.txt [hunk 1]
```diff
@@ -24,1751 +24,1751 @@ shape_id,shape_pt_lat,shape_pt_lon,shape_pt_sequence
...
-6559183,53.1858632,8.6376531,1764
-6559183,53.1885375,8.6406411,1765
-6560472,53.1947326,8.6150382,1766
-6560472,53.1925607,8.6171911,1767
-6560472,53.190696,8.6184409,1768
-6560472,53.1861406,8.6191371,1769
-6560472,53.1700892,8.6279496,1770
-6648747,53.1214837,8.7526593,1771
-6785438,53.1684836,8.6615385,1772
+19936,53.0487235,8.7423629,0
```
Comment (carocad) [inline tests/out/shapes.txt]: @mehdisadeghi was this a bug? I just realized that the shape_id changed but the shape_pt_sequence kept growing in the previous output :(

Comment (mehdisadeghi) [reply tests/out/shapes.txt]: @carocad yes it was. Previously I was creating one giant shape out of all existing stops!

Comment (carocad) [reply tests/out/shapes.txt]: oh good catch !!

**Comment by mehdisadeghi** [comment]
@carocad I fixed the dummy generator to produce hours larger than 23. Google`s `transitfeed` also approves your guess, it is valid in GTFS!

**Comment by mehdisadeghi** [comment]
Hey @carocad do you think is there anything we can improve in this PR?

**Comment by mehdisadeghi** [comment]
Here is the latest `trasitfeed` output for your convenience:

[validation-results.zip](https://github.com/hiposfer/osmtogtfs/files/1571766/validation-results.zip)

**Comment by mehdisadeghi** [comment]
@carocad  here is the latest `trasitfeed` output for your convenience:

[validation-results.zip](https://github.com/hiposfer/osmtogtfs/files/1571884/validation-results.zip)

It has 31 errors and 145 warnings. Eliminating single stop trips decreased the number of warnings from 15457 to only 145!

**Pull request merged**